### ISSUE TITLE
Can't get 'IS_AUTOINCREMENT' value from DatabaseMetaData 

### ISSUE THREAD

**Issue by downgoon** [xerial/sqlite-jdbc#218]
# Can't get 'IS_AUTOINCREMENT' value from DatabaseMetaData 

## case table in ``auto_incr.db``

``` sql
CREATE TABLE people (
 person_id INTEGER PRIMARY KEY AUTOINCREMENT,
 first_name text NOT NULL,
 last_name text NOT NULL
);
```

>the primary key ``person_id`` is **AUTOINCREMENT**

## testing code

``` java
import java.sql.Connection;
import java.sql.DatabaseMetaData;
import java.sql.DriverManager;
import java.sql.ResultSet;

public class SqliteJdbcAutoIncr {

	public static void main(String[] args) throws Exception {
		Class.forName("org.sqlite.JDBC");
		Connection conn = DriverManager.getConnection("jdbc:sqlite:auto_incr.db");

		DatabaseMetaData dbmeta = conn.getMetaData();

		// Listing Tables
		String catalog = null;
		String schemaPattern = null;
		String tableNamePattern = null;
		String[] types = null;

		ResultSet tables = dbmeta.getTables(catalog, schemaPattern, tableNamePattern, types);
		while (tables.next()) { // for each table

			final String tableName = tables.getString(3);

			// Listing Columns in a Table
			String columnNamePattern = null;
			ResultSet columns = dbmeta.getColumns(catalog, schemaPattern, tableName, columnNamePattern);

			System.out.println("table: " + tableName);
			System.out.println("----------");
			
			while (columns.next()) {
				String columnName = columns.getString(4);
				// int columnType = columns.getInt(5);

				boolean isAutoIncrement = columns.getBoolean("IS_AUTOINCREMENT");
				String autoIncString = columns.getString("IS_AUTOINCREMENT");

				System.out.println(
						String.format("\tcol: %s isAutoIncrement: %s %s", columnName, isAutoIncrement, autoIncString));
			}

			System.out.println();

		}

	}

}

```

>Exact ``IS_AUTOINCREMENT``  from metadata
``` java
boolean isAutoIncrement = columns.getBoolean("IS_AUTOINCREMENT");
String autoIncString = columns.getString("IS_AUTOINCREMENT");
```

## output result

```
table: people
----------
	col: person_id isAutoIncrement: false NO
	col: first_name isAutoIncrement: false NO
	col: last_name isAutoIncrement: false NO

table: sqlite_sequence
----------
	col: name isAutoIncrement: false NO
	col: seq isAutoIncrement: false NO
```

**'IS_AUTOINCREMENT'  value is NOT true **

>col: person_id isAutoIncrement: false NO

## sqlite-jdbc version

``` xml
<dependency>
	<groupId>org.xerial</groupId>
	<artifactId>sqlite-jdbc</artifactId>
	<version>3.16.1</version>
</dependency>
```

**Comment by doe300** [comment]
Running a short test in [SQLFiddle](http://sqlfiddle.com/#!5/08d40/1) shows, that the `PRAGMA table_info('people')` statement, which SQLite-JDBC uses to determine the column properties does not list `AUTOINCREMENT` (anymore?).

So we need another way of determining, whether a column is auto incremented. One approach I could find would be querying the table `sql_master`, e.g. `SELECT sql FROM sqlite_master WHERE name = 'people'`, which in the above [SQLFiddle](http://sqlfiddle.com/#!5/08d40/7) returns:

    CREATE TABLE people ( person_id INTEGER PRIMARY KEY AUTOINCREMENT, first_name text NOT NULL, last_name text NOT NULL )

Where we could extract the `AUTOINCREMENT` or `PRIMARY KEY` keywords.

**Comment by sualeh** [comment]
Fixed in 3.20.1
